### DISCUSSION TITLE
error crash when start on windows

### DISCUSSION THREAD

**Discussion by nuv4** [emqx/emqx#5070] [Q&A]
error crash when service start on windows error kernel?
=INFO REPORT==== 21-Jun-2021::17:34:33.337000 ===
Protocol 'inet_tcp': register/listen error: econnrefused

=SUPERVISOR REPORT==== 21-Jun-2021::17:34:33.337000 ===
    supervisor: {local,net_sup}
    errorContext: start_error
    reason: {'EXIT',nodistribution}
    offender: [{pid,undefined},
               {id,net_kernel},
               {mfargs,{net_kernel,start_link,
                                   [['emqx_maint_8228@127.0.0.1',longnames],
                                    false]}},
               {restart_type,permanent},
               {shutdown,2000},
               {child_type,worker}]
=CRASH REPORT==== 21-Jun-2021::17:34:33.337000 ===
  crasher:
    initial call: net_kernel:init/1
    pid: <0.81.0>
    registered_name: []
    exception exit: {error,badarg}
      in function  gen_server:init_it/6 (gen_server.erl, line 358)
    ancestors: [net_sup,kernel_sup,<0.47.0>]
    message_queue_len: 0
    messages: []
    links: [#Port<0.3>,<0.78.0>]
    dictionary: [{longnames,true}]
    trap_exit: true
    status: running
    heap_size: 987
    stack_size: 27
    reductions: 11320
  neighbours:

escript: exception error: no match of right hand side value 
                 {error,
                     {{shutdown,
                          {failed_to_start_child,net_kernel,
                              {'EXIT',nodistribution}}},
                      {child,undefined,net_sup_dynamic,
                          {erl_distribution,start_link,
                              [['emqx_maint_8228@127.0.0.1',longnames],false]},
                          permanent,1000,supervisor,
                          [erl_distribution]}}}
  in function  nodetool__escript__1624__289671__152000__4:'-main/1-fun-4-'/1 (C:\emqx\bin\nodetool, line 29)
  in call from nodetool__escript__1624__289671__152000__4:do_with_ret/3 (C:\emqx\bin\nodetool, line 148)
  in call from nodetool__escript__1624__289671__152000__4:main/1 (C:\emqx\bin\nodetool, line 26)
  in call from escript:run/2 (escript.erl, line 758)
  in call from escript:start/1 (escript.erl, line 277)
  in call from init:start_em/1 (init.erl, line 1109)
  in call from init:do_boot/3 (init.erl, line 817)

**Comment by zmstone** [comment]
hi @nuv4 
which EMQ X version are you running?
and what's your windows version ?
did you compile emqx from source? or did you download the package ?

**Reply by nuv4** [reply]
hi
windows version 10
emqx 4.05 download package
and MQTTX 1.3.3

**Reply by zmstone** [reply]
Could you try 4.3 (4.3.4 is the latest) instead?

**Reply by nuv4** [reply]
i tried 4.3 but error is the same
version windows 10 home 20h2
build s.o. 19042.1052

**Reply by zmstone** [reply]
sorry for the late reply.
do you have both old and new version of emqx unziped into the same directory?

and can you try to execute `emqx/erts-<VSN>/bin/erl -name testnode@127.0.0.1 -boot emqx/releases/4.3.4/start_clean` to see if you can start a regular Erlang node with `@127.0.0.1` in its name
